# **PROJETO ANÁLISE ESTATÍSTICA DO ENEM 2019**

Este projeto analisa os resultados do ENEM no estado de São Paulo do ano de 2019.

Os dados foram extraídos do site do INEP:

http://inep.gov.br/microdados

In [7]:
import numpy as np
import pandas as pd

## serve para mostar o driver
### from google.colab import drive
### drive.mount('/content/drive')

In [30]:
dados = pd.read_csv('/content/drive/MyDrive/microdados_enem_2019_sp.csv',sep=';', encoding='iso-8859-1')
# encoding: codificação de caracteres, normalmente utiliza-se o iso-8859-1, utf-8, latin-1)

## Extração e Limpeza dos dados

In [31]:
dados.head()

NU_INSCRICAO  CO_MUNICIPIO_RESIDENCIA  ... NU_NOTA_REDACAO  Q025
0     1,90E+11                  3546801  ...           940.0     B
1     1,90E+11                  3538204  ...             NaN     A
2     1,90E+11                  3550308  ...           880.0     A
3     1,90E+11                  3548708  ...           560.0     B
4     1,90E+11                  3549409  ...             NaN     B

[5 rows x 44 columns]

In [32]:
# O número total de inscritos em São Paulo foi de 813772, sendo de 15,97% de todo o Brasil que foi de 5095308 inscritos. 
dados.shape

(813772, 44)

In [ ]:
# Análise dos tipos de atributos.
# object: strings
# int64: inteiros
# float64: reais
# complex: complexos
dados.dtypes

In [18]:
# vamos fazer uma limpeza
# vamos eiminar o codigo do municipio de residência, já que temos o numero do municipio de residência
# para isso creamos un novo data frame chamado "dados1"
dados1 = dados.drop(columns=['CO_MUNICIPIO_RESIDENCIA'])

In [34]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  ...  NU_NOTA_REDACAO Q025
0     1,90E+11            Santa Isabel  ...            940.0    B
1     1,90E+11             Pinhalzinho  ...              NaN    A
2     1,90E+11               São Paulo  ...            880.0    A
3     1,90E+11   São Bernardo do Campo  ...            560.0    B
4     1,90E+11    São Joaquim da Barra  ...              NaN    B

[5 rows x 40 columns]

In [ ]:
# Agora eliminamos mais de uma coluna (no final tem que colocar a vírgula)
dados1 = dados1.drop(columns=['CO_UF_RESIDENCIA','SG_UF_RESIDENCIA','CO_MUNICIPIO_NASCIMENTO'], )

In [38]:
dados1.head(40)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  ...  NU_NOTA_REDACAO Q025
0      1,90E+11            Santa Isabel  ...            940.0    B
1      1,90E+11             Pinhalzinho  ...              NaN    A
2      1,90E+11               São Paulo  ...            880.0    A
3      1,90E+11   São Bernardo do Campo  ...            560.0    B
4      1,90E+11    São Joaquim da Barra  ...              NaN    B
5      1,90E+11              Embu-Guaçu  ...            560.0    B
6      1,90E+11     São José dos Campos  ...              NaN    A
7      1,90E+11     São José dos Campos  ...              NaN    B
8      1,90E+11               Guarulhos  ...            880.0    B
9      1,90E+11               São Paulo  ...              NaN    B
10     1,90E+11   São José do Rio Preto  ...            580.0    A
11     1,90E+11                  Sumaré  ...              NaN    B
12     1,90E+11                  Franca  ...            580.0    B
13     1,90E+11               São Paulo  ...              NaN    B
14     1,90E+11                   Bauru  ...              NaN    B
15     1,90E+11             Santo André  ...            540.0    B
16     1,90E+11                Campinas  ...              NaN    A
17     1,90E+11               São Paulo  ...            900.0    B
18     1,90E+11             Santo André  ...            520.0    B
19     1,90E+11               São Paulo  ...              NaN    B
20     1,90E+11                  Santos  ...            560.0    B
21     1,90E+11                Panorama  ...              NaN    A
22     1,90E+11              Jaguariúna  ...            500.0    A
23     1,90E+11               São Paulo  ...              NaN    A
24     1,90E+11   São Bernardo do Campo  ...              NaN    A
25     1,90E+11                     Poá  ...              NaN    B
26     1,90E+11             Carapicuíba  ...              NaN    A
27     1,90E+11                   Guará  ...            680.0    B
28     1,90E+11                 Cajamar  ...            660.0    B
29     1,90E+11               São Paulo  ...              NaN    A
30     1,90E+11                  Osasco  ...            520.0    A
31     1,90E+11               São Paulo  ...              NaN    B
32     1,90E+11               São Paulo  ...            560.0    B
33     1,90E+11               São Paulo  ...              NaN    B
34     1,90E+11               São Paulo  ...              NaN    A
35     1,90E+11               São Paulo  ...              NaN    B
36     1,90E+11                Campinas  ...            680.0    B
37     1,90E+11               São Paulo  ...            880.0    B
38     1,90E+11     São José dos Campos  ...              NaN    B
39     1,90E+11                  Osasco  ...              NaN    B

[40 rows x 40 columns]

In [39]:
# Cancelamos outras colunas que não seram necessárias para nosso estudo
dados1 = dados1.drop(columns=['NO_MUNICIPIO_NASCIMENTO','CO_UF_NASCIMENTO','SG_UF_NASCIMENTO',
                              'TP_ANO_CONCLUIU','TP_ENSINO','CO_MUNICIPIO_ESC','CO_UF_ESC','SG_UF_ESC'],)

In [40]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  ...  NU_NOTA_REDACAO Q025
0     1,90E+11            Santa Isabel  ...            940.0    B
1     1,90E+11             Pinhalzinho  ...              NaN    A
2     1,90E+11               São Paulo  ...            880.0    A
3     1,90E+11   São Bernardo do Campo  ...            560.0    B
4     1,90E+11    São Joaquim da Barra  ...              NaN    B

[5 rows x 32 columns]

In [41]:
dados1 = dados1.drop(columns=['TP_DEPENDENCIA_ADM_ESC','TP_LOCALIZACAO_ESC','TP_SIT_FUNC_ESC'],)

In [42]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  ...  NU_NOTA_REDACAO Q025
0     1,90E+11            Santa Isabel  ...            940.0    B
1     1,90E+11             Pinhalzinho  ...              NaN    A
2     1,90E+11               São Paulo  ...            880.0    A
3     1,90E+11   São Bernardo do Campo  ...            560.0    B
4     1,90E+11    São Joaquim da Barra  ...              NaN    B

[5 rows x 29 columns]

In [43]:
dados1.shape

(813772, 29)

Corrigindo erros nas notas

In [44]:
# Nas notas individuais o python não reconhecio, já que as notas do enem são de 0 a 1000
# para isso dividimos entre 10
dados1.loc[:,'NU_NOTA_CN'] /= 10

In [45]:
dados1.head(12)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  ...  NU_NOTA_REDACAO Q025
0      1,90E+11            Santa Isabel  ...            940.0    B
1      1,90E+11             Pinhalzinho  ...              NaN    A
2      1,90E+11               São Paulo  ...            880.0    A
3      1,90E+11   São Bernardo do Campo  ...            560.0    B
4      1,90E+11    São Joaquim da Barra  ...              NaN    B
5      1,90E+11              Embu-Guaçu  ...            560.0    B
6      1,90E+11     São José dos Campos  ...              NaN    A
7      1,90E+11     São José dos Campos  ...              NaN    B
8      1,90E+11               Guarulhos  ...            880.0    B
9      1,90E+11               São Paulo  ...              NaN    B
10     1,90E+11   São José do Rio Preto  ...            580.0    A
11     1,90E+11                  Sumaré  ...              NaN    B

[12 rows x 29 columns]

In [46]:
dados1.loc[:,'NU_NOTA_CH'] /= 10
dados1.loc[:,'NU_NOTA_LC'] /= 10
dados1.loc[:,'NU_NOTA_MT'] /= 10

In [47]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  ...  NU_NOTA_REDACAO Q025
0     1,90E+11            Santa Isabel  ...            940.0    B
1     1,90E+11             Pinhalzinho  ...              NaN    A
2     1,90E+11               São Paulo  ...            880.0    A
3     1,90E+11   São Bernardo do Campo  ...            560.0    B
4     1,90E+11    São Joaquim da Barra  ...              NaN    B

[5 rows x 29 columns]

Renomeando Variáveis (colunas)

In [48]:
dados1 = dados1.rename(columns={'NU_NOTA_REDACAO': 'NOTA_REDACAO'})

In [54]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE  ...  COMP5  NOTA_REDACAO  INTERNET
0     1,90E+11            Santa Isabel     22  ...  200.0         940.0         B
1     1,90E+11             Pinhalzinho     19  ...    NaN           NaN         A
2     1,90E+11               São Paulo     17  ...  200.0         880.0         A
3     1,90E+11   São Bernardo do Campo     19  ...  100.0         560.0         B
4     1,90E+11    São Joaquim da Barra     37  ...    NaN           NaN         B

[5 rows x 29 columns]

In [50]:
dados1 = dados1.rename(columns={'NU_NOTA_CN': 'NOTA_CN','NU_NOTA_CH': 'NOTA_CH',
                                'NU_NOTA_LC': 'NOTA_LC','NU_NOTA_MT': 'NOTA_MT'})

In [51]:
dados1 = dados1.rename(columns={'NU_NOTA_COMP1': 'COMP1','NU_NOTA_COMP2': 'COMP2',
                                'NU_NOTA_COMP3': 'COMP3','NU_NOTA_COMP4': 'COMP4',
                                'NU_NOTA_COMP5':'COMP5'})

In [53]:
dados1 = dados1.rename(columns={'NU_IDADE': 'IDADE','TP_SEXO': 'SEXO',
                                'TP_COR_RACA': 'RACA','Q025': 'INTERNET',
                                'TP_ESCOLA':'ESCOLA'})

Renomeando Registros (Linhas)

In [55]:
# Os numero na tabela raça indica i tipo de cor, e vamos a trocae esses número pela palabra
dados1["RACA"] = dados1["RACA"].replace({0:"nao_declarado", 1:"branca", 2:"preta", 3:"parda", 4:"amarela", 5:"indigena"})

In [58]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE  ...  COMP5  NOTA_REDACAO INTERNET
0     1,90E+11            Santa Isabel     22  ...  200.0         940.0        B
1     1,90E+11             Pinhalzinho     19  ...    NaN           NaN        A
2     1,90E+11               São Paulo     17  ...  200.0         880.0        A
3     1,90E+11   São Bernardo do Campo     19  ...  100.0         560.0        B
4     1,90E+11    São Joaquim da Barra     37  ...    NaN           NaN        B

[5 rows x 29 columns]

In [ ]:
# OUTRA MANEIRA
# dados1['RACA'] = dados1['RACA'].replace([0,1,2,3,4,5], ['Não Declarado','Branca','Preta','Parda','Amarela','Indígena'])

In [57]:
dados1["TP_LINGUA"] = dados1["TP_LINGUA"].replace({0:"Inglês", 1:"Espanhol"})

In [59]:
dados1["ESCOLA"] = dados1["ESCOLA"].replace({1:"não_respondeu", 2:"pública", 3:"privada", 4:"exterior"})

In [60]:
dados1["INTERNET"] = dados1["INTERNET"].replace({"A":"Não", "B":"Sim"})

## Análise Exploratória e Transformação dos dados

**Análise das idades**

In [61]:
dados1['IDADE'].value_counts()

17    219313
18    147548
19     81236
16     59316
20     53009
       ...  
4          1
3          1
91         1
86         1
2          1
Name: IDADE, Length: 80, dtype: int64

In [62]:
dados1['IDADE'].value_counts().sort_index()

2     1
3     1
4     1
5     2
6     2
     ..
80    2
81    3
82    2
86    1
91    1
Name: IDADE, Length: 80, dtype: int64

Alguns valores estranhos e curiosos: idades menores que 12 anos e maiores de 80 anos.

In [63]:
menores_12 = dados1.query('IDADE < 12')['NO_MUNICIPIO_RESIDENCIA'].value_counts()

In [ ]:
# Inscrições com idade abaixo de 12 anos serão excluídos devido a alta probabilidade de erro.
menores_12

São Paulo                 5
Queiroz                   1
Campinas                  1
Embu das Artes            1
Araçatuba                 1
Ribeirão Preto            1
Santana de Parnaíba       1
Bálsamo                   1
São Bernardo do Campo     1
Tambaú                    1
Auriflama                 1
Vargem Grande Paulista    1
Name: NO_MUNICIPIO_RESIDENCIA, dtype: int64

In [64]:
dados_maiores_11 = dados1.loc[dados1.IDADE > 11]

In [65]:
dados_maiores_11.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE  ...  COMP5  NOTA_REDACAO INTERNET
0     1,90E+11            Santa Isabel     22  ...  200.0         940.0      Sim
1     1,90E+11             Pinhalzinho     19  ...    NaN           NaN      Não
2     1,90E+11               São Paulo     17  ...  200.0         880.0      Não
3     1,90E+11   São Bernardo do Campo     19  ...  100.0         560.0      Sim
4     1,90E+11    São Joaquim da Barra     37  ...    NaN           NaN      Sim

[5 rows x 29 columns]

In [67]:
dados_maiores_11['IDADE'].value_counts().sort_index()

12        6
13       29
14      355
15    11583
16    59316
      ...  
80        2
81        3
82        2
86        1
91        1
Name: IDADE, Length: 73, dtype: int64

**Treineiros**

In [68]:
dados_maiores_11.query('IN_TREINEIRO == 1')['IN_TREINEIRO'].value_counts()

1    93988
Name: IN_TREINEIRO, dtype: int64

In [69]:
# vamos criar uma nova variável chamada treneiros
treineiros = dados_maiores_11.loc[dados_maiores_11.IN_TREINEIRO == 1]

In [70]:
treineiros.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE  ... COMP5  NOTA_REDACAO INTERNET
20      1,90E+11                  Santos     15  ...  80.0         560.0      Sim
21      1,90E+11                Panorama     16  ...   NaN           NaN      Não
106     1,90E+11               São Paulo     16  ...  80.0         540.0      Sim
109     1,90E+11        Águas de Lindóia     16  ...   NaN           NaN      Sim
159     1,90E+11        Águas de Lindóia     15  ...   NaN           NaN      Sim

[5 rows x 29 columns]

In [71]:
# guardamos a tabela de dados Treineiros
treineiros.to_csv('treineiros_enem_2019_sp.csv', encoding = 'iso-8859-1', index = False)

In [72]:
# Separação entre os treineiros e os vestibulandos
vestibulandos = dados_maiores_11.loc[dados_maiores_11.IN_TREINEIRO == 0]

In [73]:
vestibulandos.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE  ...  COMP5  NOTA_REDACAO INTERNET
0     1,90E+11            Santa Isabel     22  ...  200.0         940.0      Sim
1     1,90E+11             Pinhalzinho     19  ...    NaN           NaN      Não
2     1,90E+11               São Paulo     17  ...  200.0         880.0      Não
3     1,90E+11   São Bernardo do Campo     19  ...  100.0         560.0      Sim
4     1,90E+11    São Joaquim da Barra     37  ...    NaN           NaN      Sim

[5 rows x 29 columns]

In [74]:
vestibulandos.shape

(719768, 29)

**Análise das presenças**

In [79]:
# tratamento de valores ausentes NULOS
vestibulandos['TP_PRESENCA_CN'].value_counts()

1    500805
0    218693
2       270
Name: TP_PRESENCA_CN, dtype: int64

In [76]:
vestibulandos['TP_PRESENCA_CH'].value_counts()

1    533142
0    185969
2       657
Name: TP_PRESENCA_CH, dtype: int64

In [77]:
vestibulandos['TP_PRESENCA_LC'].value_counts()

1    533142
0    185969
2       657
Name: TP_PRESENCA_LC, dtype: int64

In [78]:
vestibulandos['TP_PRESENCA_MT'].value_counts()

1    500805
0    218693
2       270
Name: TP_PRESENCA_MT, dtype: int64

RESUMO DA PRESENÇA NAS PROVAS:

Ciências da Natureza: 500805 presentes, 218693 faltaram e 270 eliminados.

Ciências Humanas: 533142 presentes, 185969 faltaram e 657 eliminados.

Linguagens e Códigos: 533142 presentes, 185969 faltaram e 657 eliminados.

Matemática: 500805 presentes, 218693 faltaram e 270 eliminados.

In [80]:
vestibulandos['TP_STATUS_REDACAO'].value_counts().sort_index()

1.0    519763
2.0       913
3.0      1417
4.0      5513
6.0      3626
7.0       543
8.0       828
9.0       539
Name: TP_STATUS_REDACAO, dtype: int64

**Valores Ausentes**

In [81]:
vestibulandos['IDADE'].isnull().sum()

0

In [82]:
# Criamos uma variável 'provas' com as seguintes colunas
provas = ['NOTA_REDACAO','NOTA_CN','NOTA_MT','NOTA_LC','NOTA_CH']

In [83]:
# Registros NAN nas notas
vestibulandos[provas].isnull().sum()

NOTA_REDACAO    186626
NOTA_CN         218963
NOTA_MT         218963
NOTA_LC         186626
NOTA_CH         186626
dtype: int64

In [84]:
# Creamos uma nova variável "presenca_total" con as colunas que estão no corchete
presenca_total = ['TP_PRESENCA_CN','TP_PRESENCA_CH','TP_PRESENCA_MT','TP_PRESENCA_LC']

In [85]:
#Criado uma coluna que soma as presenças nas provas
vestibulandos['presenca_total'] = vestibulandos[presenca_total].sum(axis=1)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [86]:
vestibulandos.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  ...  INTERNET presenca_total
0     1,90E+11            Santa Isabel  ...       Sim              4
1     1,90E+11             Pinhalzinho  ...       Não              0
2     1,90E+11               São Paulo  ...       Não              4
3     1,90E+11   São Bernardo do Campo  ...       Sim              4
4     1,90E+11    São Joaquim da Barra  ...       Sim              0

[5 rows x 30 columns]

In [88]:
vestibulandos['presenca_total'].value_counts().sort_index()
# Se a soma for 0 faltou nos dois dias
# Se a soma for 2, foi em um dia
# Se a soma for 4, foi nos dois dias
# Se a soma for 6 ou 8 teve desclassificação.

0    185175
2     33726
4    500526
6       341
Name: presenca_total, dtype: int64

COMPARECIMENTO NOS DIAS DAS PROVAS:

500526 inscritos compareceram nos dois dias.

185175 inscritos faltaram nos dois dias.

33726 inscritos faltaram em um dos dias.

341 foram desclassificados.



In [89]:
#Eliminando os estudantes que foram desclassificados por terem comparecido em apenas um dos dias, isto é, ficando apenas com os que vão concorrer no vestibular.
vestibulandos_presentes = vestibulandos.loc[vestibulandos.presenca_total == 4]

In [90]:
vestibulandos_presentes['TP_PRESENCA_CN'].value_counts().sort_index()

0       583
1    499940
2         3
Name: TP_PRESENCA_CN, dtype: int64

In [91]:
vestibulandos_presentes['TP_PRESENCA_CH'].value_counts().sort_index()

0         3
1    499940
2       583
Name: TP_PRESENCA_CH, dtype: int64

In [92]:
vestibulandos_presentes['TP_PRESENCA_LC'].value_counts().sort_index()

0         3
1    499940
2       583
Name: TP_PRESENCA_LC, dtype: int64

In [93]:
vestibulandos_presentes['TP_PRESENCA_MT'].value_counts().sort_index()

0       583
1    499940
2         3
Name: TP_PRESENCA_MT, dtype: int64

In [94]:
vestibulandos_classificados = vestibulandos_presentes.loc[vestibulandos_presentes.TP_PRESENCA_CN == 1]

In [95]:
vestibulandos_classificados['TP_PRESENCA_CN'].value_counts()

1    499940
Name: TP_PRESENCA_CN, dtype: int64

In [96]:
vestibulandos_classificados['TP_PRESENCA_CH'].value_counts()

1    499940
Name: TP_PRESENCA_CH, dtype: int64

In [97]:
vestibulandos_classificados['TP_PRESENCA_LC'].value_counts()

1    499940
Name: TP_PRESENCA_LC, dtype: int64

In [98]:
vestibulandos_classificados['TP_PRESENCA_MT'].value_counts()

1    499940
Name: TP_PRESENCA_MT, dtype: int64

In [99]:
provas = ['NOTA_REDACAO','NOTA_CN','NOTA_MT','NOTA_LC','NOTA_CH']

In [100]:
vestibulandos_classificados[provas].isnull().sum()

NOTA_REDACAO    0
NOTA_CN         0
NOTA_MT         0
NOTA_LC         0
NOTA_CH         0
dtype: int64

** Notas zeros na Redação**

In [101]:
vestibulandos_classificados.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

0.0    9181
Name: NOTA_REDACAO, dtype: int64

In [102]:
vestibulandos_classificados.query('COMP1 == 0')['COMP1'].value_counts()

0.0    9188
Name: COMP1, dtype: int64

In [103]:
vestibulandos_classificados.query('COMP2 == 0')['COMP2'].value_counts()

0.0    9181
Name: COMP2, dtype: int64

In [104]:
vestibulandos_classificados.query('COMP3 == 0')['COMP3'].value_counts()

0.0    9197
Name: COMP3, dtype: int64

In [105]:
vestibulandos_classificados.query('COMP4 == 0')['COMP4'].value_counts()

0.0    9194
Name: COMP4, dtype: int64

In [106]:
vestibulandos_classificados.query('COMP5 == 0')['COMP5'].value_counts()

0.0    73330
Name: COMP5, dtype: int64

NOTAS ZEROS EM REDAÇÃO:

REDAÇÃO (final): 9181 notas zeros devido a não cumprimento das regras ou tirou zero em todos os cinco componentes.

COMPONENTE 1(Demonstrar domínio da modalidade escrita formal da Língua Portuguesa.): 9188 notas zeros somente neste item.

COMPONENTE 2 (Compreender a proposta de redação e aplicar conceitos das várias áreas de conhecimento para desenvolver o tema, dentro dos limites estruturais do texto dissertativo-argumentativo em prosa): 9181. 

COMPONENTE 3 (Selecionar, relacionar, organizar e interpretar informações, fatos, opiniões e argumentos em defesa de um ponto de vista): 9197 notas zeros somente neste item.

COMPONENTE 4 (Demonstrar conhecimento dos mecanismos linguísticos necessários para a construção da argumentação.): 9194 notas zeros somente neste item.

COMPONENTE 5 (Elaborar proposta de intervenção para o problema abordado, respeitando os direitos humanos): 73330 notas zeros somente neste item.


In [107]:
vestibulandos_classificados.query('NOTA_CH == 0')['NOTA_CH'].value_counts()

0.0    343
Name: NOTA_CH, dtype: int64

In [108]:
vestibulandos_classificados.query('NOTA_CN == 0')['NOTA_CN'].value_counts()

0.0    36
Name: NOTA_CN, dtype: int64

In [109]:
vestibulandos_classificados.query('NOTA_LC == 0')['NOTA_LC'].value_counts()

0.0    135
Name: NOTA_LC, dtype: int64

In [110]:
vestibulandos_classificados.query('NOTA_MT == 0')['NOTA_MT'].value_counts()

0.0    56
Name: NOTA_MT, dtype: int64

In [111]:
vestibulandos_classificados.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

0.0    9181
Name: NOTA_REDACAO, dtype: int64

NOTAS ZEROS DOS INSCRITOS CONCORRENTES AO VESTIBULAR:

Matemática = 56

Ciências da Natureza = 36

Linguagens e Códigos = 135

Ciências Humanas = 343

Redação = 9181


In [112]:
vestibulandos_classificados.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  ...  INTERNET presenca_total
0     1,90E+11            Santa Isabel  ...       Sim              4
2     1,90E+11               São Paulo  ...       Não              4
3     1,90E+11   São Bernardo do Campo  ...       Sim              4
5     1,90E+11              Embu-Guaçu  ...       Sim              4
8     1,90E+11               Guarulhos  ...       Sim              4

[5 rows x 30 columns]

In [113]:
vestibulandos_classificados = vestibulandos_classificados.drop(columns=['IN_TREINEIRO','presenca_total'], )

In [114]:
vestibulandos_classificados.shape

(499940, 28)

## **Salvando (Exportando) o Dataframe Tratado**

In [115]:
vestibulandos_classificados.to_csv('enem_2019_tratado.csv', encoding = 'iso-8859-1', index = False)